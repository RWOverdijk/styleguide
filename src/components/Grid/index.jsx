import React, { PureComponent } from 'react';
import './style.scss';

class Grid extends PureComponent {
  render() {
    return <div className="row">{this.props.children}</div>;
  }
}

export default Grid;
