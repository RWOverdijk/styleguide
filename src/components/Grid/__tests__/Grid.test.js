import React from 'react';
import renderer from 'react-test-renderer';
import Grid from '../';

describe("<Grid>", () => {
  it("renders grid", () => {
    const tree = renderer.create(
      <Grid>My Grid</Grid>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <Grid className="my-class">My Grid</Grid>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <Grid data-id="0">My Grid</Grid>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
})
