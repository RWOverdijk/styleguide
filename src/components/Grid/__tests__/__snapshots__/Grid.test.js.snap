// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid> renders grid 1`] = `
<div
  className="row"
>
  My Grid
</div>
`;

exports[`<Grid> renders with specific attributes 1`] = `
<div
  className="row"
  data-id="0"
>
  My Grid
</div>
`;

exports[`<Grid> renders with specific className 1`] = `
<div
  className="row my-class"
>
  My Grid
</div>
`;
